tools/multipathsecret/multipathsecret/vaultconfig.py

```diff
@@ -0,0 +1,125 @@
+import hvac
+import json
+import os
+from dataclasses import dataclass, field
+from typing import List, Dict, Optional
+
+
+@dataclass
+class Keyset:
+    '''Contains "accessor" and "id" tokens for a Vault path.
+    '''
+    accessor: str
+    id: str
+
+
+@dataclass
+class Enclave:
+    '''Maps a Vault path to 'read' and 'write' Keysets.  The 'Enclave' is
+    simply a particular vault path with its own Keysets, e.g.
+    'k8s_operator/nublado.lsst.codes'
+    '''
+    name: str
+    read: Keyset
+    write: Keyset
+
+
+@dataclass
+class VaultConfig:
+    '''Contains the vault address (a URL represented as a string), the
+    secret to be added, which is a dict mapping strings to strings
+    (but can be None, if you're deleting a secret from a path), the
+    rendered-to-memory JSON document representing all the vault paths
+    and tokens, and the list of vault paths to not update.
+
+    '''
+    vault_address: str = os.getenv('VAULT_ADDR')
+    secret: Optional[Dict[str, str]] = None
+    enclaves: List[Enclave] = field(default_factory=list)
+    skip_list: List[str] = field(default_factory=list)
+
+    def __init__(self,
+                 vault_address: Optional[str],
+                 vault_file: str,
+                 skip_list: Optional[List[str]],
+                 secret_file: Optional[str] = None):
+        if vault_address:
+            self.vault_address = vault_address
+        if secret_file:
+            self.load_secret(secret_file)
+        with open(vault_file, "r") as f:
+            vault_dict = json.load(f)
+        self.enclaves = []
+        for item in vault_dict:
+            name = list(item.keys())[0]
+            if name in skip_list:
+                continue
+            read_k = Keyset(**item[name]["read"])
+            write_k = Keyset(**item[name]["write"])
+            enclave = Enclave(name=name, read=read_k, write=write_k)
+            self.enclaves.append(enclave)
+
+    def load_secret(self, secret_file: str) -> None:
+        with open(secret_file, "r") as f:
+            self.secret = json.load(f)
+
+    def _get_write_key_for_enclave(self, enclave: Enclave) -> str:
+        '''Given a Vault path, return its write key.
+        '''
```

Comment:
Nit: This function takes an Enclave with the keys already loaded, not a Vault path.

Reply:
Done